src/main/java/com/google/sps/data/Places.java

```diff
@@ -143,4 +146,21 @@
     }
     return stream;
   }
+
+  /**
+   * This function returns the hidden gems ranked by rating (descending order)
+   * @param hiddenGems                      This takes as parameter the set of hiddenGems to be sorted. 
+   * @return ArrayList<PlacesSearchResult>  This returns an arrayList of the hidden gems ranked by rating.
+   */
+  public static ArrayList<PlacesSearchResult> getRankedHiddenGems(Set<PlacesSearchResult> hiddenGems) {
```

Comment:
Unless you're relying on something specific about arraylist probably better to return this as a List

Reply:
Yep, all List implementations are ordered.

Reply:
I have changed it to a List